### ISSUE TITLE
Tracker issue for adoption status

### ISSUE THREAD

**Issue by AkihiroSuda** [containerd/stargz-snapshotter#258]
# CRI runtimes
## containerd
- ✅ Supported since containerd 1.4.0 

## CRI-O
- TBD, will happen after [Podman](#podman) supports lazy-pulling

# High-level container engines
## Docker (Moby)
- On proposal: https://github.com/moby/moby/pull/41002

## Podman
- On proposal: https://github.com/containers/storage/pull/795 https://github.com/containers/image/pull/1109

## nerdctl
- ✅ Supported since nerdctl 0.0.1 https://github.com/AkihiroSuda/nerdctl/blob/master/docs/stargz.md

# Image builders
### Lazy-pulling base images
- ✅ Supported since BuildKit 0.8.0 https://github.com/moby/buildkit/blob/master/docs/stargz-estargz.md

### Building eStargz images
- Not possible with BuildKit, but possible by combining BuildKit, containerd, and nerdctl https://github.com/AkihiroSuda/nerdctl/blob/master/docs/stargz.md

### Lazy-pulling base images
- ❌ Won't be possible
### Building eStargz images
- ✅ Supported since Kaniko 1.4.0 https://github.com/GoogleContainerTools/kaniko/pull/1527

### Lazy-pulling base images
- TBD, will happen after [Podman](#podman) supports lazy-pulling
### Building eStargz images
- TBD

**Comment by ktock** [comment]
Thanks! :+1:

**Comment by ktock** [comment]
@chenk008

> Does Podman need to bump containers/storage version to support stargz?

Yes. You also need https://github.com/containers/image/pull/1109 and https://github.com/containerd/stargz-snapshotter/pull/301 as well.
We'll work on downstreaming these patches to Podman/CRI-O once they are merged.

Please check also the current limitation of lazy pulling feature for these runtimes (https://github.com/containers/storage/pull/795#issuecomment-799057091). Summary:
- Podman cannot export (save/push) lazily pulled layer
- Podman and CRI-O don't verify lazily-pulled chunks

**Comment by AkihiroSuda** [comment]
Linking
- https://github.com/rancher-sandbox/rancher-desktop/issues/2707

**Comment by gabrieldemarmiesse** [comment]
In the new Docker desktop, the containerd image store is in beta. I tried it and it used stargz as the default snapshotter. It's a bit slower than nerdctl but pretty cool nonetheless

**Comment by giuseppe** [comment]
@ktock what is needed for chunk validation in Podman/CRI-O/Buildah?

Isn't that part done by the snapshotter itself?

**Comment by harche** [comment]
> @ktock what is needed for chunk validation in Podman/CRI-O/Buildah?
> 
> Isn't that part done by the snapshotter itself?

/cc @AkihiroSuda

**Comment by jonathanbeber** [comment]
I believe crane should be removed as https://github.com/google/go-containerregistry/pull/1660

**Comment by sevetseh28** [comment]
does this mean that `docker pull` will lazily pull an image now? I\m trying it but I'm not sure if it's unsupported or if something is misconfigured on my end. Using `nerdctl` does lazily pull it
